include/Graph/Graph.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include <queue>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+class Layer {
+ private:
+  int id;
+  std::string name;
+  int type;
+  std::string version;
+  int numInputs;
+  int numNeurons;
+  std::vector<int> primer;
+
+ public:
+  Layer(int id1, int type1) : id(id1), type(type1) {}
+  int checkID() { return id; }
+  void In(const std::vector<int>& a) { primer = a; }
+  std::vector<int> Out() { return primer; }
+};
+
+class Graph {
+  int BiggestSize;
+  int V;
+  std::vector<Layer> layers;
+  std::vector<int> arrayV;
+  std::vector<int> arrayE;
+  int start;
+  int end;
+
+ public:
+  Graph(int vertices) : BiggestSize(vertices) {
+    if (BiggestSize < 0) {
+      throw std::out_of_range("Vertices cannot be less than zero");
+    }
+    arrayV.push_back(0);
+    V = 0;
+  }
+  void input(Layer lay, int end1, std::vector<int> vec) {
+    layers.push_back(lay);
+    arrayV.push_back(0);
+    start = lay.checkID();
+    end = end1;
+    V++;
+  }
```

Comment:
@aobolensk about this implementation: how can we deduce id, where graph ends?

Reply:
What do you mean by graph end here? If you mean the end layer, it will be denoted by layer type

Reply:
Changed now